lib/experimental/synchronization.php

```diff
@@ -22,3 +22,145 @@ function gutenberg_rest_api_init_collaborative_editing() {
+ * not in-sync with the Yjs state, it will update the Yjs document.
+ *
+ * To ensure that all clients update the Yjs state in "the same way" and
+ * produce the same Yjs update, all client must use the same Yjs-clientid. This
+ * clientid must change whenever the HTML content updates, to prevent the
+ * creation of conflicting Yjs updates.
+ *
+ * Note: Yjs has a concept of clientId that is very different from the
+ * clientIds used in the block editor. Yjs' clientIds should be unique per
+ * client (i.e. each browser tab has a different clientId) and are used for
+ * conflict-resolution.
+ *
+ * It is usually not recommended to change the clientid, as this can corrup the
+ * Yjs document and make it unusable. Please consult an expert on Yjs CRDTs
+ * before changing this approach.
+ *
+ * This approach is not ideal and may - under very specific circumstances -
+ * lead to content duplication.
+ *
+ * When multiple changes to the HTML document happen (without updating the Yjs
+ * state) while multiple collaboration-enabled clients listen to changes, it
+ * may result in content duplication.
+ *
+ * Example:
+ *
+ *   - Change 1: Paragraph 1 is added to the HTML content without updating the
+ *               Yjs document.
+ *   - Change 2: Paragraph 2 is added to the HTML content without updating the
+ *               Yjs document. This change happens immediately after change 1.
+ *               So this changes also incorporates the changeset of change 1.
+ *
+ * Result:
+ *
+ *   - Clients that see change 1 will add paragraph 1 to the Yjs document.
+ *   - Clients that see change 2 will add paragraph 1 and paragraph 2 to the
+ *     Yjs document, using a different clientid.
+ *   - In total, three paragraphs are added. The clients have no way of knowing
+ *     that change 2 incorporates changes from change 2.
+ *
+ * If content duplication happens a lot, it may be necessary to increase the
+ * debounce interval between fetching document states.
+ *
+ * A real solution would be to maintain diffs in the y:gutenberg comment when changes
+ * happen without Yjs noticing. However, such an implementation will further
+ * increase the size of the y:gutenberg comment.
+ *
+ * In practice, we can accept content duplication in some edge-cases. This
+ * is better that the status quo, which overwrites existing content and can
+ * lead to data loss.
+ */
+function filter_post_content_ydoc( $data, $postarr, $unsanitized_postarr ) {
+	if ($data['post_type'] !== 'post' and $data['post_type'] !== 'revision') {
+		return $data;
+	}
+	$gutenberg_experiments = get_option( 'gutenberg-experiments' );
+	if ( ! $gutenberg_experiments || ! array_key_exists( 'gutenberg-sync-collaboration', $gutenberg_experiments ) ) {
+		return $data;
+	}
+	$content = stripslashes($data['post_content']);
+	// generate a new clientid for updated content that can be represented as a 53bit unsigned integer (max clientid in Yjs)
+	$ynewclientid = rand(0, 9007199254740991); // $match[4];
```

Comment:
```suggestion
	$ynewclientid = wp_rand(0, 9007199254740991); // This is 2^53 – 1 which is `Number.MAX_SAFE_INTEGER` from JavaScript
```

`wp_rand`  should be a bit more random.  Also adjusted the comment so it's not unused code but instead explains the number used.